adapters/uniqueid_linux.c

```diff
@@ -25,13 +39,36 @@ UNIQUEID_RESULT UniqueId_Generate(char* uid, size_t len)
     }
     else
     {
-        uuid_t uuidVal;
-        uuid_generate(uuidVal);
 
-        /* Codes_SRS_UNIQUEID_07_001: [UniqueId_Generate shall create a unique Id 36 character long string.] */
-        memset(uid, 0, len);
-        uuid_unparse(uuidVal, uid);
-        result = UNIQUEID_OK;
+        // generate version 4 uuid
+        struct uuid unique_id;
+
+        // get random bytes
+        ssize_t bytes_got = getrandom(&unique_id, sizeof(unique_id), 0);
```

Comment:
earlier behavior of uuid_generate was to never block even if the high-quality randomness from /dev/urandom is not available or initialized, because it use to fallback on alternative algorithm.
"The uuid will be generated based on high-quality randomness from /dev/urandom, if available. If it is not available, then uuid_generate will use an alternative algorithm which uses the current time, the local ethernet MAC address (if available), and random data generated using a pseudo-random generator. "

But with flag zero to function from sys/random.h, it will block if the high quality random is not initialized. Should we fail fast by adding flag GRND_NONBLOCK?